src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -366,6 +370,35 @@ private void EmitPointerTypeSignature(PointerType type, SignatureContext context
             EmitTypeSignature(type.ParameterType, context);
         }
 
+        private void EmitFunctionPointerTypeSignature(FunctionPointerType type, SignatureContext context)
+        {
+            CorCallingConvention callingConvention = (type.Signature.Flags & MethodSignatureFlags.UnmanagedCallingConventionMask) switch
+            {
+                MethodSignatureFlags.None => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_DEFAULT,
+                MethodSignatureFlags.UnmanagedCallingConventionCdecl => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_C,
+                MethodSignatureFlags.UnmanagedCallingConventionStdCall => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_STDCALL,
+                MethodSignatureFlags.UnmanagedCallingConventionThisCall => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_THISCALL,
+                MethodSignatureFlags.CallingConventionVarargs => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_VARARG,
+                MethodSignatureFlags.UnmanagedCallingConvention => CorCallingConvention.IMAGE_CEE_CS_CALLCONV_UNMANAGED,
+                _ => throw new NotSupportedException()
+            };
+
+            if ((type.Signature.Flags & MethodSignatureFlags.Static) == 0)
+            {
+                callingConvention |= CorCallingConvention.IMAGE_CEE_CS_CALLCONV_HASTHIS | CorCallingConvention.IMAGE_CEE_CS_CALLCONV_EXPLICITTHIS;
+            }
+
```

Comment:
Does this need to emit modops / modreqs, or are those intentionally skipped here?

Reply:
Well, we probably do, I'll try to fix this but the problem is that apparently there's no pre-existing coverage for function pointer signatures in CoreCLR testing so I guess I'll need to create some dedicated tests to verify this works as expected especially as this is the first place in Crossgen2 where we'll be emitting these signature modifiers.

Reply:
Actually - if we are ever only going to use signature to create type handle at runtime, we should not need to emit the modopts/modreqs. They are not part of the type handle at runtime.

Reply:
Also, the function pointer support in reflection is broken and inconsistent in various ways today. We hope to fix that as part of https://github.com/dotnet/runtime/issues/69273 in .NET 8.

Reply:
That's exactly why I asked for a broad review, this is obviously new functionality with very limited test coverage (amounting to the one JIT test I'm re-enabling in this PR).

Reply:
We have logic to handle emission of modopts/modreqs, but that should only be needed for function signatures, and those are already handled correctly so far as I'm aware. For this support I do not believe we need to handle modreqs/modopts at all, ever.